examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
I think you misunderstand my intent.

NONE means NONE, as in NO OUTPUT at all.

the _NONE case means - all log output is completely disabled.  The intent in *THIS* case here (cli/main.c, and ncp/main.c) is to provide a *DUMMY* stub function that performs no work.  

Technically we could remove this 'dummy-stub' if every place on the platform has proper conditionals around their call to the log function that disabled/removed that call.  If it is true that all locations in the application do not call this function then there is no "call-graph-arc" into this function. We can then rely on the common compiler/linker options of:   -ffunction-section (which not all compilers support) which will remove unused functions from the code.

In the worse case scenario (some code is stupid and still calls this) - it becomes a dummy function that i a few bytes in size, in the grand scheme of things - 2 to 4 bytes out of 100K bytes is not very important.

Another scenario would be a library supplied in binary form that requires this symbol and leaves it to the function to log or not log.

Anything other then _NONE means some means of outputting debug log messages exists - and the selection where you want the output to goto the APPLICATION uart - is the second case - an example is the CERT_LOG feature, where the log must go to the cli console for certification reasons.